x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -370,7 +370,9 @@ private void finalise(int slot, AutoFollowResult result) {
                         // has a leader index uuid custom metadata entry that matches with uuid of leaderIndexMetaData variable
                         // If so then handle it differently: not follow it, but just add an entry to
                         // AutoFollowMetadata#followedLeaderIndexUUIDs
-                        leaderIndicesToFollow.add(leaderIndexMetaData.getIndex());
+                        if (TransportPutFollowAction.canFollow(clusterAlias, leaderIndexMetaData) == null) {
```

Comment:
Initially I wanted to do this check in `checkAutoFollowPattern(...)` method, so that we could fail or log when an index can't be followed, but then I realized that we would do this for every auto follow run and that would be way to noisy. So I think this is the right place.